### DISCUSSION TITLE
USB port - camera sync

### DISCUSSION THREAD

**Discussion by MariaDaniela24** [bonsai-rx/bonsai#1255] [Q&A]
Hey there!

I am trying to use Bonsai to record 4 cameras at the same time on one computer.

(For now, I am just doing the test with two cameras.)
When I connect the cameras directly to the USB ports of the computer everything works.
When I connect them to a USB port adapter (to have 4 entries instead of 2), it doesn't work anymore.

Do you know how can I solve this problem?

Thank you in advance for all the help,

Maria

**Comment by bruno-f-cruz** [comment]
This is unlikely to be a problem with Bonsai. Can you detect the 4 cameras when using the adapter with any other software?

**Comment by benefron** [comment]
This is probably because all of it is streamed to the same USB port and has limited bandwidth. How fast are the cameras? Which USB port is it (2,3,3.1)? Connecting each camera to its own port is good practice to ensure enough bandwidth for all the data streaming.

**Comment by MariaDaniela24** [comment]
Thank you for all your answers!

I will keep the good practices.
